src/Pages/FieldInfoPage.js

```diff
@@ -0,0 +1,244 @@
+  }
+}));
+
+
+/**
+ * Define columns for Fields Info Table
+ * * Need to add last two entries for edit and delete icons to be visible
+ */
+const columnData = [
+  { id: 'id', label: 'Id', minWidth: 5 },
+  { id: 'name', label: 'Field Name', minWidth: 30 },
+  {
+    id: 'location',
+    label: 'Locality',
+    minWidth: 30
+  },
+  {
+    id: 'area',
+    label: 'Area',
+    minWidth: 10
+  },
+  {
+    id: 'edit',
+    type: 'icon',
+    minWidth: 5,
+    align: 'left',
+    label: ''
+  },
+  {
+    id: 'delete',
+    type: 'icon',
+    minWidth: 5,
+    align: 'left',
+    label: ''
+  }
+];
+
+/**
+ * Component to show Field Information
+ * @returns form allowing location selection
+ * @returns Table to show field info
+ */
+function FieldInfoPage() {
+  const classes = useStyles();
+  const [locations, setLocations] = useState([]);
+  const [locality, setLocality] = useState('');
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [alertSeverity, setAlertSeverity] = useState('');
+  const [fieldsData, setFields] = useState([]);
+
+  useEffect(() => {
+    getLocations()
+      .then(setLocations)
+      .catch((e) => {
+        console.log('Fetching all locations failed', e);
+      })
+  }, []);
+
+  /**
+   * Function to fetch all field information using API 
+   * updates fieldsData if call is successful
+   * shows alert in case call fails
+   */
+  const getFieldData = () => {
+    getAllFields()
+      .then((fields) => {
+        setFields(fields.content);
+      }).catch((e) => {
+        console.log('Fetching all fields failed', e);
+        showAlert(`Fetching fields failed`, 'error');
+      })
+  };
+
+  useEffect(() => {
+    getFieldData();
+  }, []);
+
+  /**
+   * Function to create and return row data for binding to table
+   * 
+   */
+  const getRowData = () => {
+    if (fieldsData && fieldsData.length) {
+      let data = fieldsData.map((obj) => {
+        return {
+          id: obj.id,
+          name: obj.identifier,
+          location: obj.location.displayStr,
+          area: obj.area
+        };
+      });
+      return data;
+    }
+    else return [];
+  };
+
+  /**
+   * Handler called on closing alert 
+   * @param {*} _event 
+   * @param {*} reason 
+   */
+  const handleAlertClose = (_event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setAlertStatus(false);
+  };
+
+  /**
+   * Handler called when user selects location from dropdown
+   * sets the locality value to the selection
+   * @param {*} object 
+   */
+  const handleLocationChange = ({ target }) => {
+    const { name, value } = target;
+    if (name === 'locality') {
+      setLocality(value);
+    }
+  };
+
+  /**
+   * Function to fetch fields for a selected ocation using API
+   * updates fieldsData if call is successful
+   * shows alert in case call fails 
+   */
+  const fetchFieldsForLocation = () => {
+    getAllFields(locality)
```

Comment:
Instead of updating the existing method to take location as input, can we please add a separate method `findFieldsByLocation`. Seems simpler and cleaner that way.

Reply:
Generally, we add query parameters as part of same function. The number of query parameters might increase in future and adding a separate function for each of them does not look like a proper solution.

Reply:
Yes, but the method existing here is like GET_ALL which doesn't usually take any query param. 
Another way to look at it is, a finder usually served paginated results, which is what we would eventually do, as the number of fields could be really large. The table component should also be showing X rows per page. GET_ALL doesn't have pagination usually. Hence the suggestion to model it as a finder separately.

Reply:
done